### ISSUE TITLE
V 5.7.5 breaks  Binding Resolution

### ISSUE THREAD

**Issue by j3rrey** [laravel/lumen-framework#849]
- Lumen Version: 5.7.5
- PHP Version: 7.2

### Description:
I've just done an upgrade from 5.5 to 5.6 to 5.7 until 5.7.4 everything works as expected, as soon as I pull in 5.7.5.
Php Artisan gives me "Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable."

Example:
```
php artisan

In Container.php line 945:

  Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable.
```

### Steps To Reproduce:

upgrade from 5.5 to 5.6 to 5.7 and fin ally from 5.7.4 to 5.7.5.

I'm checking all the code changes and what could possibly be causing this atm.

Maybe you guys have an idea ? 

Here is the stacktrace: 
```
#[2018-11-15 15:30:50] local.ERROR: Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /project/vendor/illuminate/container/Container.php:945
#Stack trace:
#0 project/vendor/illuminate/container/Container.php(785): Illuminate\Container\Container->notInstantiable('Illuminate\\Cont...')
#1 project/vendor/illuminate/container/Container.php(658): Illuminate\Container\Container->build('Illuminate\\Cont...')
#2 project/vendor/illuminate/container/Container.php(609): Illuminate\Container\Container->resolve('Illuminate\\Cont...', Array)
#3 project/vendor/laravel/lumen-framework/src/Application.php(260): Illuminate\Container\Container->make('Illuminate\\Cont...', Array)
#4 project/vendor/illuminate/container/Container.php(1222): Laravel\Lumen\Application->make('Illuminate\\Cont...')
#5 project/vendor/laravel/lumen-framework/src/Console/Kernel.php(223): Illuminate\Container\Container->offsetGet('Illuminate\\Cont...')
#6 project/vendor/laravel/lumen-framework/src/Console/Kernel.php(117): Laravel\Lumen\Console\Kernel->reportException(Object(Illuminate\Contracts\Container\BindingResolutionException))
#7 project/artisan(43): Laravel\Lumen\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 {main} {"exception":"[object] (Illuminate\\Contracts\\Container\\BindingResolutionException(code: 0): Target [Illuminate\\Contracts\\Debug\\ExceptionHandler] is not instantiable. at project/vendor/illuminate/container/Container.php:945)
#[stacktrace]
#0 project/vendor/illuminate/container/Container.php(785): Illuminate\\Container\\Container->notInstantiable('Illuminate\\\\Cont...')
#1 project/vendor/illuminate/container/Container.php(658): Illuminate\\Container\\Container->build('Illuminate\\\\Cont...')
#2 project/vendor/illuminate/container/Container.php(609): Illuminate\\Container\\Container->resolve('Illuminate\\\\Cont...', Array)
#3 project/vendor/laravel/lumen-framework/src/Application.php(260): Illuminate\\Container\\Container->make('Illuminate\\\\Cont...', Array)
#4 project/vendor/illuminate/container/Container.php(1222): Laravel\\Lumen\\Application->make('Illuminate\\\\Cont...')
#5 project/vendor/laravel/lumen-framework/src/Console/Kernel.php(223): Illuminate\\Container\\Container->offsetGet('Illuminate\\\\Cont...')
#6 project/vendor/laravel/lumen-framework/src/Console/Kernel.php(117): Laravel\\Lumen\\Console\\Kernel->reportException(Object(Illuminate\\Contracts\\Container\\BindingResolutionException))
#7 project/artisan(43): Laravel\\Lumen\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#8 {main}
```

**Comment by driesvints** [comment]
The actual problem is probably something deeper. Apparently something's being resolved from the container which can't be resolved. I've systematically try to disable and re-add service providers and bootstrapped services to see what's causing the actual problem. Try die/dumping through your `bootstrap/app.php` file to see if you can pinpoint the problem.

**Comment by driesvints** [comment]
These are all the changes between said versions btw: https://github.com/laravel/lumen-framework/compare/v5.7.4...v5.7.5

**Comment by j3rrey** [comment]
This one is breaking it for me 
https://github.com/laravel/lumen-framework/commit/425907aa4ba7540bce44960470362eaf594e9248

Now investigating why.

**Comment by driesvints** [comment]
This was changed to https://github.com/laravel/lumen-framework/commit/d973359b1714536c36cf989278b7be8eff3e30ad in the meantime but unreleased yet. Can you check if that fixes your problem? I'll release a new version if it does.

**Comment by j3rrey** [comment]
Yes It does! 
```
php artisan
"self::class: Laravel\Lumen\Application"
"static::class: App\Application"
```

**Comment by driesvints** [comment]
Released v5.7.6. Thanks for checking 👍

**Comment by j3rrey** [comment]
"laravel/lumen-framework": "dev-master#d973359b1714536c36cf989278b7be8eff3e30ad",
is working as expected.
Thx and have a nice evening.
